public/app/plugins/panel/canvas/InlineEdit.tsx

```diff
@@ -0,0 +1,197 @@
+import { setOptionImmutably } from 'app/features/dashboard/components/PanelEditor/utils';
+
+import { InstanceState, activePanelSubject } from './CanvasPanel';
+import { getElementEditor } from './editor/elementEditor';
+import { getLayerEditor } from './editor/layerEditor';
+
+type Props = {
+  onClose?: () => void;
+};
+
+const OFFSET = 8;
+
+export const InlineEdit = ({ onClose }: Props) => {
+  const activePanel = useObservable(activePanelSubject);
+  const instanceState = activePanel?.panel.context?.instanceState;
+  const theme = useTheme();
+  const btnInlineEdit = document.querySelector('[data-btninlineedit]')!.getBoundingClientRect();
+  const ref = useRef<HTMLDivElement>(null);
+  const styles = getStyles(theme);
+  const inlineEditKey = 'inlineEditPanel';
+
+  const defaultMeasurements = { width: 350, height: 400 };
+  const defaultX = btnInlineEdit.x - btnInlineEdit.width + OFFSET;
+  const defaultY = -OFFSET - defaultMeasurements.height;
+
+  const savedPlacement = store.getObject(inlineEditKey, {
+    x: defaultX,
+    y: defaultY,
+    w: defaultMeasurements.width,
+    h: defaultMeasurements.height,
+  });
+  const [measurements, setMeasurements] = useState<Dimensions2D>({ width: savedPlacement.w, height: savedPlacement.h });
+  const [placement, setPlacement] = useState({ x: savedPlacement.x, y: savedPlacement.y });
+
+  const pane = useMemo(() => {
+    const state: InstanceState = instanceState;
+    if (!state) {
+      return new OptionsPaneCategoryDescriptor({ id: 'root', title: 'root' });
+    }
+
+    const supplier = (builder: PanelOptionsEditorBuilder<any>, context: StandardEditorContext<any>) => {
+      builder.addNestedOptions(getLayerEditor(instanceState));
+
+      const selection = state.selected;
+      if (selection?.length === 1) {
+        const element = selection[0];
+        if (!(element instanceof GroupState)) {
+          builder.addNestedOptions(
+            getElementEditor({
+              category: [`Selected element (${element.options.name})`],
+              element,
+              scene: state.scene,
+            })
+          );
+        }
+      }
+    };
+
+    return getOptionsPaneCategoryDescriptor({}, supplier);
+  }, [instanceState]);
+
+  const onDragStop = (event: any, dragElement: any) => {
+    setPlacement({ x: dragElement.x, y: dragElement.y });
+    saveToStore(dragElement.x, dragElement.y, measurements.width, measurements.height);
+  };
+
+  const onResizeStop = (event: SyntheticEvent<Element, Event>, data: ResizeCallbackData) => {
+    const { size } = data;
+    setMeasurements({ width: size.width, height: size.height });
+    saveToStore(placement.x, placement.y, size.width, size.height);
+  };
+
+  const saveToStore = (x: number, y: number, width: number, height: number) => {
+    store.setObject(inlineEditKey, { x: x, y: y, w: width, h: height });
+  };
+  return (
+    <Draggable handle="strong" onStop={onDragStop} position={{ x: placement.x, y: savedPlacement.y }}>
```

Comment:
This can all live in a portal -- so that it can be relative to the window rather than the panel

Reply:
The current shift when you drop seems to be the offset of the page header, so that will likely get fixed too